tree_sitter/__init__.py

```diff
@@ -2,8 +2,13 @@
 
 import enum
 from ctypes import c_void_p, cdll
-from distutils.ccompiler import new_compiler
-from distutils.unixccompiler import UnixCCompiler
+try:
+    from distutils.ccompiler import new_compiler
+    from distutils.unixccompiler import UnixCCompiler
+except ImportError:
+    from setuptools._distutils.ccompiler import new_compiler
+    from setuptools._distutils.unixccompiler import UnixCCompiler
```

Comment:
I'm not a huge fan of this method, is there one import supported by 3.7-3.12? I'm not too familiar w/ what changed in 3.12

Reply:
`setuptools>=60.0.0` includes a vendored copy of `distutils` ([ref](https://setuptools.pypa.io/en/latest/deprecated/distutils-legacy.html)), so the only change necessary is to depend on `setuptools` for Python >= 3.12. When `setuptools` is installed, importing `distutils` will automatically use the vendored copy from `setuptools`:

```pycon
>>> import distutils.unixccompiler
>>> distutils.unixccompiler
<module 'distutils.unixccompiler' from '/usr/lib/python3.11/site-packages/setuptools/_distutils/unixccompiler.py'>
```

Reply:
I'm fine adding setuptools as a dependency, unless there's downsides to that

Reply:
I'm updating the PR then. Setting the dependency as `extra` is another discussion I suppose :)